src/Client.php

```diff
@@ -200,6 +200,24 @@ public function registerDefaultCallbacks()
         return $this;
     }
 
+    /**
+     * Register a middleware object to the pipeline.
+     *
+     * @param callable $middleware
+     *
+     * @return $this
+     */
+    public function registerMiddleware(callable $middleware)
+    {
+        if (is_callable($middleware)) {
+            $this->pipeline->pipe($middleware);
+        } else {
+            syslog(LOG_WARNING, 'Middleware '.get_class($middleware).' could not be added to the pipeline');
```

Comment:
This code is not reachable, and should be deleted.

Reply:
You're correct, I was just verifying the earlier versions of PHP had the same behaviour

Reply:
They should do. Callable was introduced in php 5.4 as a typehint, and the behavior has remained the same since.

Reply:
I'm not sure what kind of error is thrown in earlier versions of PHP, in PHP 7 it's quite explicitly a TypeError, any idea for PHP 5.6?

Reply:
https://3v4l.org/FnNTN